Migrate ESLint config file to new syntax

Seems I've left some bits out of https://github.com/srcery-colors/srcery.sh/pull/458 and it needs some updating.

Guide: https://eslint.org/docs/latest/use/configure/migration-guide#configuring-language-options

@roosta some of the options in `eslint.config.js`, do you have a relation to them?  Like the custom parser, etc.  I'm asking as it require some changes to transfer them over.

Cleaned it down to:
```js
// ESLint Config

import js from "@eslint/js";
import eslintConfigPrettier from "eslint-config-prettier";
// import babelParser from "@babel/eslint-parser";

const config = [
  js.configs.recommended,
  // {
  //   files: ["**/*.js", "**/*.mjs"],
  //   languageOptions: {
  //     parser: babelParser,
  //   },
  // },
  eslintConfigPrettier,
];

//   {
//   env: {
//     browser: true,
//     node: true,
//     es2021: true,
//   },
//   extends: ["eslint:recommended", "prettier"],
//   parserOptions: {
//     ecmaVersion: 13,
//     sourceType: "module",
//   },
//   rules: {
//     "no-unused-vars": "warn",
//   },
// };

export default config;
```
---

I actually found a few bugs:

```

/Users/birgerjn/Sites/srcery.sh/src/main.js
  54:3  error  Duplicate key 'lua'      no-dupe-keys
  69:3  error  'window' is not defined  no-undef
```

--- a/eslint.config.js
+++ b/eslint.config.js
@@ -1,20 +1,32 @@
 // ESLint Config
 
-const config = {
-  env: {
-    browser: true,
-    node: true,
-    es2021: true,
+import babelParser from "@babel/eslint-parser";
+import eslintConfigPrettier from "eslint-config-prettier";
+import globals from "globals";
+import js from "@eslint/js";
+
+const config = [
+  {
+    ignores: ["**/dist"],
   },
-  extends: ["eslint:recommended", "prettier"],
-  parser: "@babel/eslint-parser",
-  parserOptions: {
-    ecmaVersion: 13,
-    sourceType: "module",
+
+  {
+    languageOptions: {
+      globals: {
+        ...globals.browser,
+        ...globals.es2021,
+        ...globals.node,
+      },
+      parser: babelParser,
+      parserOptions: {
+        requireConfigFile: false,
+      },
+    },
   },
-  rules: {
-    "no-unused-vars": "warn",
-  },
-};
+
+  js.configs.recommended,
+
+  eslintConfigPrettier,
+];
 
 export default config;

--- a/package.json
+++ b/package.json
@@ -11,6 +11,7 @@
   "license": "MIT",
   "author": "Srcery Team",
   "type": "module",
+  "browser": true,
   "scripts": {
     "build": "vite build --emptyOutDir",
     "clean": "rm -f src/github.json",
@@ -26,9 +27,13 @@
     "highlight.js": "^11.8.0"
   },
   "devDependencies": {
+    "@babel/core": "^7.22.10",
+    "@babel/eslint-parser": "^7.22.10",
+    "@types/eslint": "^8.44.2",
     "autoprefixer": "^10.4.14",
     "eslint": "^8.47.0",
     "eslint-config-prettier": "^9.0.0",
+    "globals": "^13.21.0",
     "htmlhint": "^1.1.4",
     "lodash": "^4.17.21",
     "octokit": "^3.1.0",

--- a/src/main.js
+++ b/src/main.js
@@ -51,7 +51,7 @@ const languages = {
     html: hljs.highlight(luatxt, { language: "lua" }).value,
     name: "Lua",
   },
-  lua: {
+  rust: {
     html: hljs.highlight(rusttxt, { language: "rust" }).value,
     name: "Rust",
   },

